automerge/src/autocommit.rs

```diff
@@ -321,28 +321,50 @@ impl Transactable for AutoCommit {
     /// - The object does not exist
     /// - The key is the wrong type for the object
     /// - The key does not exist in the object
-    fn set<P: Into<Prop>, V: Into<Value>>(
+    fn set<P: Into<Prop>, V: Into<ScalarValue>>(
         &mut self,
         obj: &ExId,
         prop: P,
         value: V,
-    ) -> Result<Option<ExId>, AutomergeError> {
+    ) -> Result<(), AutomergeError> {
```

Comment:
Can we modify the `set` and `insert` methods to only accept a `ScalarValue` and likewise check in the `set_object` and `insert_object` that the arguments are in fact objects? Otherwise I think we run the risk of people calling `insert_object` with a non-object value and getting back something they think is an object ID but which is actually just an opid?

Reply:
Oh wait, I think that's already covered by the `Into`s, my bad.

Reply:
Yep, I think those are covered